samples/hello/src/main/java/com/ea/orbit/samples/memoize/MemoizeInterceptor.java

```diff
@@ -0,0 +1,36 @@
+package com.ea.orbit.samples.memoize;
+
+import com.ea.orbit.actors.IAddressable;
+import com.ea.orbit.actors.runtime.IRuntime;
+import com.ea.orbit.actors.runtime.InvokeInterceptor;
+import com.ea.orbit.concurrent.Task;
+import net.jodah.expiringmap.ExpiringMap;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class MemoizeInterceptor implements InvokeInterceptor {
+
+    ExpiringMap<String, Task> memoizeMap = ExpiringMap.builder().variableExpiration().build();
+
+    @Override
+    public Task invoke(IRuntime runtime, IAddressable toReference, Method m, boolean oneWay, int methodId, Object[] params) {
+        if (m.isAnnotationPresent(Memoize.class))
```

Comment:
minor, but instead of invoking isAnnotationPresent and then getAnnotation, you could simply call getAnnotation and then null check the result.